### PR TITLE
version and readme changes

### PR DESCRIPTION
**Pull request by VarunSharma15**
### Description
Made changes into Readme.md regarding compatibility in between versions of dbt-core, dbt-utils, teradata-utils and dbt_teradata_adapter.
Made version changes wherever necessary.

### Checklist
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [ ] This PR includes tests, or tests are not required/relevant for this PR
 - [ ] I have updated the `CHANGELOG.md` with information about my change

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,11 @@ At a minimum, you need to specify `host`, `user`, `password`, `schema` (database
 | 1.0.0.x           | ❌           | ✅          | ✅          | ✅          | ❌          |
 |1.1.0.x            | ❌           | ✅          | ✅          | ✅          | ✅          |
 
+## Version compatibility
```
Comment (SatishChGit) [inline README.md:42]: Can you change the title to dbt dependent package version compatibility

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
-## dbt-teradata 1.1.0a
+## dbt-teradata 1.2.0a
 
 ### Features
+Connection Retries option added.
+Implemented Grants feature.
```
Comment (SatishChGit) [inline CHANGELOG.md:5]: Better to include these feature comments once implementation completed.

**Review by SatishChGit — CHANGES_REQUESTED** [review]
Looks good. Minor review comments added. Once addressed, will approve

File: README.md (hunk 1 above)
Comment (VarunSharma15) [reply README.md]: Made the changes.

File: CHANGELOG.md (hunk 2 above)
Comment (VarunSharma15) [reply CHANGELOG.md]: Sure, edited the file.

**Review by tallamohan — COMMENTED** [review]
Looks Good after the changes "changed title and Changelog.md"

**Review by tallamohan — APPROVED** [review]
Looks Good.

File: .bumpversion.cfg [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.1.0a
```
Comment (adamtwo) [inline .bumpversion.cfg:2]: We don't have to change this with each release. It's just a marker for bumpversion to do its thing during the release process. We could keep it at 1.0.0a at all times. Changing this string is error-prone. If we miss a file where this placeholder string is referenced, it will produce a faulty release. It's ok to change it to 1.2.0a in this release, but I recommend we go back to 1.0.0a in the next release and leave it alone :).

File: README.md [hunk 4]
```diff
@@ -39,6 +39,11 @@ At a minimum, you need to specify `host`, `user`, `password`, `schema` (database
 | 1.0.0.x           | ❌           | ✅          | ✅          | ✅          | ❌          |
 |1.1.0.x            | ❌           | ✅          | ✅          | ✅          | ✅          |
 
+##  dbt dependent packages version compatibility
```
Comment (adamtwo) [inline README.md:42]: don't we need a row in the python compatibility matrix for 1.2.0.x? Maybe it will be introduced with a different PR?

File: README.md [hunk 5]
```diff
@@ -135,6 +140,27 @@ my-teradata-db-profile:
       port: <port>
 ```
 
+### Retries
```
Comment (adamtwo) [inline README.md:143]: This update needs to be published to https://docs.getdbt.com/reference/warehouse-setups/teradata-setup and possibly https://docs.getdbt.com/reference/resource-configs/teradata-configs . This is done by submitting a PR to https://github.com/dbt-labs/docs.getdbt.com . For example, see https://github.com/dbt-labs/docs.getdbt.com/pull/1027

File: dbt/adapters/teradata/connections.py [hunk 6]
```diff
@@ -175,6 +177,26 @@ def open(cls, connection):
         # Save the transaction mode
         cls.TMODE = credentials.tmode
 
+        logger.debug("host: {}",credentials.server);
+        logger.debug("retries: {}",credentials.retries);
+
+        if credentials.retries > 0:
```
Comment (adamtwo) [inline dbt/adapters/teradata/connections.py:183]: It would be nice to have a test for this. So far, we have only done integration tests. An integration test for a retry is probably not feasible, but maybe it's time to introduce a unit test framework?

**Review by adamtwo — CHANGES_REQUESTED** [review]

File: .bumpversion.cfg (hunk 3 above)
Comment (VarunSharma15) [reply .bumpversion.cfg:2]: Sure Adam, We will revert this change in the future release.
Thanks for pointing this.

File: README.md (hunk 4 above)
Comment (VarunSharma15) [reply README.md:42]: Yes Adam, we have added the python compatibility for 1.2.0.x in the other PR.

File: README.md (hunk 5 above)
Comment (tallamohan) [reply README.md:143]: Sure, Adam. I will create a branch in https://github.com/dbt-labs/docs.getdbt.com and make the required changes.

File: dbt/adapters/teradata/connections.py (hunk 6 above)
Comment (tallamohan) [reply dbt/adapters/teradata/connections.py:183]: Ok, Adam. We will plan to introduce the unit test framework.

**Review by adamtwo — APPROVED** [review]

**Review by SatishChGit — APPROVED** [review]
Looks good with latest changes
